packages/libs/react-ui/src/components/Button/NewButton.tsx

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable @kadena-dev/no-eslint-disable */
+
+import { useObjectRef } from '@react-aria/utils';
+import type { RecipeVariants } from '@vanilla-extract/recipes';
+import type {
+  ComponentProps,
+  ForwardedRef,
+  MouseEvent,
+  ReactNode,
+} from 'react';
+import React, { forwardRef, useCallback } from 'react';
+import type { AriaButtonProps } from 'react-aria';
+import { useButton } from 'react-aria';
+import { Loading } from '../Icon/System/SystemIcon';
+import { button, spinner } from './NewButton.css';
+
+type Variants = Omit<NonNullable<RecipeVariants<typeof button>>, 'onlyIcon'>;
```

Comment:
Does this make all recipe variants required props? If so, I don't think we should remove `NonNullable` and have the variants default to what is set in the recipe so users are not required to add every prop

Reply:
no it doesn't, it is a workaround for vanilla extract types to make it work with `Omit`